kernel/src/hil/kv_store.rs

```diff
@@ -0,0 +1,109 @@
+//! Interface for KV Stores
+
+use crate::returncode::ReturnCode;
+
+#[derive(Clone, Copy, PartialEq)]
+pub enum Operation {
+    None,
+    Init,
+    GetKey,
+    AppendKey,
+    InvalidateKey,
+    GarbageCollect,
+}
+
+/// The type of keys, this should define the output size of the digest
+/// operations.
+pub trait KeyType: Eq + Copy + Clone + Sized + AsRef<[u8]> + AsMut<[u8]> {}
+
+/// Implement this trait and use `set_client()` in order to receive callbacks.
+pub trait Client<'a, K: KeyType> {
+    /// This callback is called when any operation is complete.
+    ///
+    /// `operation`: The operation that completed
+    /// `result`: Nothing on success, 'ReturnCode' on error
+    /// `key`: A key buffer is one exists
+    /// `value`: A value buffer is one exists
+    /// `buf`: A buf buffer is one exists
+    fn operation_complete(
+        &'a self,
+        operation: Operation,
+        result: Result<(), ReturnCode>,
+        key: Option<&'static mut K>,
+        value: Option<&'static mut [u8]>,
+        buf: Option<&'static mut [u8]>,
+    );
+}
+
+pub trait KVStore<K: KeyType> {
+    /// The type of the hashed key. For example '[u8; 64]'.
+    type KeyType;
+
```

Comment:
I guess the `set_client` method could be part of the HIL, as done with the other HILs (such as UART).

Reply:
Added